### PR TITLE
Fold ShardGetService creation away from Guice  into IndexShard

### PR DESCRIPTION
**Pull request by s1monw**
it's always acccessed via IndexShard and has crazy circular dependencies or
rather had. It just makes IndexShard ctor bigger for no reason.

### CODE REVIEW

**Comment by rjernst** [comment]
LGTM

File: core/src/main/java/org/elasticsearch/index/get/ShardGetService.java [hunk 1]
```diff
@@ -238,7 +223,7 @@ public GetResult innerGet(String type, String id, String[] gFields, boolean real
                             value = source.source.length();
                         } else {
                             if (searchLookup == null) {
-                                searchLookup = new SearchLookup(mapperService, fieldDataService, new String[]{type});
+                                searchLookup = new SearchLookup(mapperService, null, new String[]{type});
```
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/get/ShardGetService.java]: Is it ok to omit passing the fieldDataService?

Comment (rjernst) [inline core/src/main/java/org/elasticsearch/index/get/ShardGetService.java]: This is for document GET right? So it shouldn't be using fielddata, only translog or stored fields?

Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/get/ShardGetService.java]: Good point, I was thinking about fielddata_fields but we can't get them anyway if a doc is only in the translog...

Comment (s1monw) [inline core/src/main/java/org/elasticsearch/index/get/ShardGetService.java]: yeah that was my intention :)

**Pull request merged**